### PR TITLE
Add support for enabling/disabling netscaler features

### PR DESCRIPTION
**Pull request by davemcdonnell**

### CODE REVIEW

File: lib/puppet/type/netscaler_feature.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  apply_to_device
+
+  ensurable
+  
+  newparam(:name, :namevar => true) do
+    desc 'Feature name'
+    
+    validate do |value|
+      if ! [:wl,:sp,:lb,:cs,:cr,:sc,:cmp,:pq,:ssl,:gslb,:hdosp,:cf,:ic,:sslvpn,:aaa,:ospf,:rip,:bgp,:rewrite,:ipv6pt,:appfw,:responder,:htmlinjection,:push,:appflow,:cloudbridge,:isis,:ch,:appqoe,:diskcaching,:vpath,:contentaccelerator,:rise,:feo].any?{ |s| s.casecmp(value.to_sym) == 0 }
+        fail ArgumentError, "Valid options: wl, sp, lb, cs, cr, sc, cmp, pq, ssl, gslb, hdosp, cf, ic, sslvpn, aaa, ospf, rip, bgp, rewrite, ipv6pt, appfw, responder, htmlinjection, push, appflow, cloudbridge, isis, ch, appqoe, diskcaching, vpath, contentaccelerator, rise, feo"
```
Comment (hunner) [inline lib/puppet/type/netscaler_feature.rb]: Hmm, I wonder if we should make these the english names instead of the REST API names? What do you think?

**Comment by davemcdonnell** [comment]
Netscaler admins will probably know them as their alias, LB, etc. 
eg enable ns feature LB

It would be nice to have it handle both, Im not sure how I would handle prefetching etc with a namevar that is aliased to one real resource. I guess I could put two copies in when I prefetch (one with alias, one with english name) but that would probably break puppet semantics around single declarations:
netscaler_feature { 'Load Balancing': ensure => present, }
netscaler_feature { 'LB': ensure => absent, }

**Comment by jonnytpuppet** [comment]
@davemcdonnell I think all @Hunner meant was that we use :feature_name instead of :name in the type. It's literally a one word change :)

**Comment by hunner** [comment]
@jonnytpuppet Nope nope, I did mean changing all the wl, sb, lb, etc. strings to the english sentences used in the GUI.

@davemcdonnell So... do you think most netscaler devs would prefer the arguments used in the command-line rather than the gui? For the most part we have been modeling the module after the GUI's verbiage when there is a discrepancy between the GUI and the CLI (eg, `td` vs `traffic_domain` as a parameter name). What do you think?

**Comment by davemcdonnell** [comment]
True netscaler admins use the command line :).

I think there is validity to both approaches. Users that have existing automation/scripts will probably be using the CLI/Rest names, and others approaching puppet automation fresh may be more comfortable with the GUI names.

I'll see if I can come up with something using title_patterns to support both rest and english names, with puppet resource returning the english names to be consistent with that approach in the module.

**Comment by hunner** [comment]
If title_patterns doesn't work, `self.prefetch` could match up resources based on either their cli or gui name, rather than just the name directly.

Or `name`'s munge method could munge them to the cli name.

**Comment by davemcdonnell** [comment]
Resubmitted in different pull request

**Pull request closed without merging**